File: integration-tests/TestAPI-bindings/c-sharp/Cricut.TestAPI.Tests/ClassTests.cs
using System;
using Xunit;

namespace Cricut.TestAPI.Tests {
    public class ClassTests {
        [Fact]
        void testEmptyStruct() {
            var a1 = EmptyClass1.Create();
            var a2 = EmptyClass1.Create();
            Assert.Equal("Blarg!", a1.GetBlarg());
            Assert.Equal("<wibble>*Wobble*", a1.GetWibbledyWobbledyTimeyWhimey());
            Assert.Equal("Shme! Hand me my hook.", a1.Shme());
            Assert.Equal("Go Gorp!", a1.Gorpers());

            var b1 = EmptyClass2.Make();
            Assert.Equal("Gralb!", b1.GetBlorg());
            Assert.Equal("<timey>*Whimey*", b1.GetWibble());
            Assert.Equal("Shme? What's that ticking sound?", b1.Shmee());
            Assert.Equal("Stop Sreprog!", b1.Gorp());

            Assert.Equal(a1.ToString(), a2.ToString());

            Assert.NotEqual(a1.ToString(), b1.ToString());
        }
    }
}

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/ClassTests.kt
package com.cricut.testapi

import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

internal class ClassTests {
    companion object {
        const val DEBUG_PRINTS = true
        @BeforeAll
        @JvmStatic
        // To get procId for attaching lldb debugger to
        fun beforeAll() {
            val procId = ProcessHandle.current().pid()
            if (DEBUG_PRINTS) {
                println("procId: $procId")
            }
        }
    }
    @Test
    fun testEmptyClass() {
        val a1 = EmptyClass1.create()
        val a2 = EmptyClass1.create()
        assertEquals("Blarg!", a1.blarg)
        assertEquals("<wibble>*Wobble*", a1.wibbledyWobbledyTimeyWhimey)
        assertEquals("Shme! Hand me my hook.", a1.shme())
        assertEquals("Go Gorp!", a1.Gorpers())

        assertEquals(a1, a2)
        assertEquals(a1.hashCode(), a2.hashCode())

        val b1 = EmptyClass2.make()
        assertEquals("Gralb!", b1.blorg)
        assertEquals("<timey>*Whimey*", b1.wibble)
        assertEquals("Shme? What's that ticking sound?", b1.shmee())
        assertEquals("Stop Sreprog!", b1.gorp())

        assertNotEquals(a1, b1)
        assertNotEquals(a1.hashCode(), b1.hashCode())

        assertEquals("TestAPI.EmptyClass", a1.toString())
        assertEquals("TestAPI.EmptyClass2", b1.toString())
    }
}
